### PR TITLE
Test coverage increased

### PR DESCRIPTION
**Pull request by rjt-gupta**
Changes this PR proposes -

1. Bumped test coverage of `snare_helpers.py` to 100%. Overall coverage increased to 75%.
2. `.coverage` is added to `.gitignore` file.

Refs - #187

### CODE REVIEW

File: snare/tests/test_parse_timeout.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import unittest
+from argparse import ArgumentTypeError
+
+from snare.utils.snare_helpers import parse_timeout, str_to_bool
+
+
```
Comment (rnehra01) [inline snare/tests/test_parse_timeout.py]: Remove extra line

File: snare/tests/test_parse_timeout.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+class TestParseTimeout(unittest.TestCase):
+
+    def setUp(self):
+        self.v = None
+
+    def test_parse_timeout(self):
+
```
Comment (rnehra01) [inline snare/tests/test_parse_timeout.py]: Same here.

File: snare/tests/test_parse_timeout.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        assert parse_timeout('20H') == 72000
+        assert parse_timeout('10M') == 600
+        assert parse_timeout('1D') == 86400
+
+        assert parse_timeout('24Y') == 86400  # Default 24H format is used.
+
+    def test_str_to_bool(self):
```
Comment (rnehra01) [inline snare/tests/test_parse_timeout.py]: This can be put separately.

File: snare/cloner.py [hunk 4]
```diff
@@ -19,7 +19,7 @@ def __init__(self, root, max_depth, css_validate):
         self.max_depth = max_depth
         self.moved_root = None
         if len(self.root.host) < 4:
-            sys.exit('invalid taget {}'.format(self.root.host))
+            sys.exit('invalid target {}'.format(self.root.host))
```
Comment (rnehra01) [inline snare/cloner.py]: Make the PR focus on a certain task, please put it in a different PR.

File: snare/utils/snare_helpers.py [hunk 5]
```diff
@@ -63,12 +63,12 @@ def add_meta_tag(page_dir, index_page, config):
         main_page = main.read()
     soup = BeautifulSoup(main_page, 'html.parser')
 
-    if (google_content and soup.find("meta", attrs={"name": "google-site-verification"}) is None):
+    if google_content and soup.find("meta", attrs={"name": "google-site-verification"}) is None:
```
Comment (rnehra01) [inline snare/utils/snare_helpers.py]: Put it in a different PR and the rest too.

File: snare/utils/snare_helpers.py [hunk 6]
```diff
@@ -63,12 +63,12 @@ def add_meta_tag(page_dir, index_page, config):
...
         google_meta = soup.new_tag('meta')
         google_meta.attrs['name'] = 'google-site-verification'
         google_meta.attrs['content'] = google_content
         soup.head.append(google_meta)
-    if (bing_content and soup.find("meta", attrs={"name": "msvalidate.01"}) is None):
+    if bing_content and soup.find("meta", attrs={"name": "msvalidate.01"}) is None:
```
Comment (rnehra01) [inline snare/utils/snare_helpers.py]: Same here.

**Comment by rjt-gupta** [comment]
Okayy, Will do the changes asap!

**Comment by rjt-gupta** [comment]
Updated the requested changes.

File: snare/tests/test_add_meta_tag.py [hunk 7]
```diff
@@ -28,5 +28,8 @@ def test_add_meta_tag(self):
         assert(soup.find("meta", attrs={"name": "google-site-verification"}) and
                soup.find("meta", attrs={"name": "msvalidate.01"}))
 
+        config['WEB-TOOLS'] = dict(google='', bing='')
```
Comment (rnehra01) [inline snare/tests/test_add_meta_tag.py]: Usually, different input conditions for a method are tested using separate test methods, so you can have a method name like `test_add_meta_tag_with_empty_tags` or anything else.

File: snare/tests/test_str_to_bool.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+from argparse import ArgumentTypeError
+from snare.utils.snare_helpers import str_to_bool
+
+
+class TestStrToBool(unittest.TestCase):
+
+    def setUp(self):
+        self.v = None
+
+    def test_str_to_bool(self):
```
Comment (rnehra01) [inline snare/tests/test_str_to_bool.py]: Same for this case. `True`, `False` and `Error` separate.

File: snare/tests/test_parse_timeout.py [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import unittest
+from snare.utils.snare_helpers import parse_timeout
+
+
+class TestParseTimeout(unittest.TestCase):
+
+    def test_parse_timeout(self):
+        assert parse_timeout('20H') == 72000
```
Comment (rnehra01) [inline snare/tests/test_parse_timeout.py]: Can you write it as `20*60*60`? It will make more sense. And update the others too.

File: snare/tests/test_str_to_bool.py [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+        self.v = 'true'
+        assert str_to_bool(self.v) is True
+        self.v = 'false'
+        assert str_to_bool(self.v) is False
+
+        with self.assertRaises(ArgumentTypeError):
+            str_to_bool('twz')
```
Comment (rnehra01) [inline snare/tests/test_str_to_bool.py]: You are using `self.v` for the previous two cases, do the same for this too.

File: snare/tests/test_parse_timeout.py (hunk 9 above)
Comment (rjt-gupta) [reply snare/tests/test_parse_timeout.py]: Yes, makes more sense. Will do the changes

**Comment by rjt-gupta** [comment]
Changes Done.

File: snare/tests/test_parse_timeout.py [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import unittest
+from snare.utils.snare_helpers import parse_timeout
+
+
+class TestParseTimeout(unittest.TestCase):
+
+    def test_parse_timeout(self):
+        assert parse_timeout('20H') == 20*60*60
+        assert parse_timeout('10M') == 10*60
+        assert parse_timeout('1D') == 1*86400
```
Comment (rnehra01) [inline snare/tests/test_parse_timeout.py]: `24*60*60`

Comment (rjt-gupta) [reply snare/tests/test_parse_timeout.py]: Yes, sorry about this.
